docs/setup-log4j2.asciidoc

```diff
@@ -42,90 +48,123 @@ Instead of the usual `<PatternLayout/>`, use `<EcsLayout serviceName="my-app"/>`
+
+|`eventDataset`
+|String
+|`${serviceName}.log`
+|Sets the `event.dataset` field used by the machine learning job of the Logs app to look for anomalies in the log rate.
+
+|includeMarkers`
+|boolean
+|`false`
+|Log https://logging.apache.org/log4j/2.0/manual/markers.html[Markers] as {ecs-ref}/ecs-base.html[`tags`]
+
+|`stackTraceAsArray``
+|boolean
+|`false`
+|Serializes the {ecs-ref}/ecs-error.html[`error.stack_trace`] as a JSON array where each element is in a new line to improve readability.
+ Note that this requires a slightly more complex <<setup-stack-trace-as-array, Filebeat configuration>>.
+
+|`includeOrigin`
+|boolean
+|`false`
+|If `true`, adds the {ecs-ref}/ecs-log.html[`log.origin.file.name`],
+ {ecs-ref}/ecs-log.html[`log.origin.file.line`] and {ecs-ref}/ecs-log.html[`log.origin.function`] fields.
+ Note that you also have to set `includeLocation="true"` on your loggers and appenders if you are using the async ones.
+|===
 
 To include any custom field in the output, use following syntax:
 
-```xml
+[source,xml]
+----
   <EcsLayout>
     <KeyValuePair key="additionalField1" value="constant value"/>
     <KeyValuePair key="additionalField2" value="$${ctx:key}"/>
   </EcsLayout>
-```
+----
 
 Custom fields are included in the order they are declared. The values support [lookups](https://logging.apache.org/log4j/2.x/manual/lookups.html).
 
-## Structured logging
+[float]
+=== Structured logging
 
 By leveraging log4j2's `MapMessage` or even by implementing your own `MultiformatMessage` with JSON support,
 you can add additional fields to the resulting JSON.
 
 Example:
 
-```java
+[source,java]
+----
 logger.info(new StringMapMessage()
     .with("message", "Hello World!")
     .with("foo", "bar"));
-``` 
+----
 
 If Jackson is on the classpath, you can also use an `ObjectMessage` to add a custom object the resulting JSON.
 
-```java
+[source,java]
+----
 logger.info(new ObjectMessage(myObject));
-```
+----
 
 The `myObject` variable refers to a custom object which can be serialized by a Jackson `ObjectMapper`.
 
 Using either will merge the object at the top-level (not nested under `message`) of the log event if it is a JSON object.
 If it's a string, number boolean or an array, it will be converted into a string and added as the `message` property.
 The conversion is done in order to avoid mapping conflicts as `message` is typed as a string in the Elasticsearch mapping.
 
-### Tips
-It's recommended to use existing [ECS fields](https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html).
+[float]
+==== Tips
+It's recommended to use existing [ECS fields]({ecs-ref}/ecs-field-reference.html).
 
 If there is no appropriate ECS field,
 consider prefixing your fields with `labels.`, as in `labels.foo`, for simple key/value pairs.
 For nested structures consider prefixing with `custom.` to make sure you won't get conflicts if ECS later adds the same fields but with a different mapping.
```

Comment:
```suggestion
If there is no appropriate ECS field,
consider prefixing your fields with `labels.`, as in `labels.foo`, for simple key/value pairs.
For nested structures, consider prefixing with `custom.`. This approach protects against conflicts in case ECS later adds the same fields but with a different mapping.
```